rmf_fleet_adapter/src/rmf_fleet_adapter/agv/FleetUpdateHandle.cpp

```diff
@@ -718,23 +635,50 @@ void FleetUpdateHandle::Implementation::dispatch_request_cb(
 auto FleetUpdateHandle::Implementation::is_valid_assignments(
   Assignments& assignments) const -> bool
 {
-  std::unordered_set<std::string> executed_tasks;
```

Comment:
The original intention of this function is to ensure that none of the tasks in the previously generated assignments has begun execution. The was achieved efficiently in the original implementation. With your changes, you call `is_queued()` inside a nested `for` loop which in turn runs two `for` loops to compile a list of `queued_tasks`. Repeating this for assignment per agent is highly inefficient. Please retain the original implementation.

Reply:
Yes, this still preserves the intention in a clearer way. The potential problem by using `get_executed_tasks` is if an active task is being moved back to the queue (future implementation), it will still be considered in the `get_executed_tasks`.